### PR TITLE
Link prediction notebook and dining dataset added

### PR DESCRIPTION
**Pull request by acostadon**
Adding notebook to run all link predictions and a new small dataset with string names for nodes

### CODE REVIEW

File: datasets/dining_prefs.mtx [hunk 1]
```diff
@@ -0,0 +1,54 @@
+%%MatrixMarket matrix coordinate pattern symmetric
```
Comment (BradReesWork) [inline datasets/dining_prefs.mtx:1]: Why do we need MTX file

File: python/cugraph/cugraph/datasets/metadata/dining_prefs.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+name: dining_prefs
+file_type: .csv
+author: J.L. Moreno
```
Comment (nv-rliu) [inline python/cugraph/cugraph/datasets/metadata/dining_prefs.yaml:3]: Between `file_type` and `author`, we should add a `description` field that briefly explains what this graph represents

File: datasets/dining_prefs.csv [hunk 3]
```diff
@@ -0,0 +1,52 @@
+Ada Cora 1
```
Comment (BradReesWork) [inline datasets/dining_prefs.csv:1]: I thought that dataset was being saved in the cloud and accessed via:  https://data.rapids.ai/cugraph/datasets/dining_prefs.csv

so why do we needed that datafile here?

File: python/cugraph/cugraph/datasets/metadata/dining_prefs.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+url:  https://data.rapids.ai/cugraph/datasets/dining_prefs.csv
```
Comment (nv-rliu) [inline python/cugraph/cugraph/datasets/metadata/dining_prefs.yaml:4]: All the metadata fields are standardized now so the `url` field is at the very end.

File: python/cugraph/cugraph/datasets/metadata/dining_prefs.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  - string
+  - string
+  - int
+has_loop: true
+is_directed: true
+is_multigraph: false
+is_symmetric: true
+number_of_edges: 52
+number_of_nodes: 26
+number_of_lines: 52
```
Comment (nv-rliu) [inline python/cugraph/cugraph/datasets/metadata/dining_prefs.yaml:23]: The other YAML files no longer include `number_of_lines` in their metadata. Just # nodes and # edges

**Review by BradReesWork — CHANGES_REQUESTED** [review]

**Pull request closed without merging**